### PR TITLE
Initial RSS backend for perceval

### PR DESCRIPTION
**Pull request by acs**

### CODE REVIEW

File: perceval/backends/core/rss.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+        return ts.timestamp()
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a RSS item.
+
+        This backend only generates one type of item which is
+        'item'.
+        """
+        return 'item'
```
Comment (sduenas) [inline perceval/backends/core/rss.py]: Is this the category for these items? In other places you talk about entries, not about items. Item is so generic, I think.

File: perceval/backends/core/rss.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    :param url: URL of rss node: https://item.opnfv.org/ci
+
+    :raises HTTPError: when an error occurs doing the request
+    """
+
+    def __init__(self, url):
+        self.url = url
+
+    def get_entries(self):
+        """ Retrieve all entries from a RSS feed
```
Comment (sduenas) [inline perceval/backends/core/rss.py]: One-line docstrings follow the format `"""Your text here"""`

File: perceval/backends/core/rss.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+        """
+        req = requests.get(self.url)
+        req.raise_for_status()
+        return req.text
+
```
Comment (sduenas) [inline perceval/backends/core/rss.py]: Leave one black line here too.

File: tests/test_rss.py [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+# Hack to make sure that tests import the right packages
+# due to setuptools behaviour
+sys.path.insert(0, '..')
+pkg_resources.declare_namespace('perceval.backends')
+
+from perceval.cache import Cache
+from perceval.errors import CacheError
+from perceval.backends.core.rss import (RSS,
+                                            RSSCommand,
```
Comment (sduenas) [inline tests/test_rss.py]: Bad identation here.

File: tests/test_rss.py [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+
+RSS_FEED_URL = 'http://example.com/rss'
+
+requests_http = []
+
+def read_file(filename, mode='r'):
+    with open(filename, mode) as f:
+        content = f.read()
+    return content
+
```
Comment (sduenas) [inline tests/test_rss.py]: Leave two blank lines between classes.

File: tests/test_rss.py [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+        """Test if it returns True when has_caching is called"""
+
+        self.assertEqual(RSS.has_caching(), True)
+
+    def test_has_resuming(self):
+        """Test if it returns False when has_resuming is called"""
+
+        self.assertEqual(RSS.has_resuming(), False)
+
+    def __configure_http_server(self):
```
Comment (sduenas) [inline tests/test_rss.py]: This method can go outside of this class.

File: tests/test_rss.py [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+        rss = RSS(RSS_FEED_URL)
+        entries = [item for item in rss.fetch()]
+        self.assertEqual(len(entries), 30)
+
+        # Test metadata
+        expected = [('98572defb3a652afbfdfe96517edefb88a22dcfa', 1481044620.0),
+                    ('e3b0d7463fca0c47d82debc3ddc37d8906f77548', 1480955040.0),
+                    ('28bca39353ff0825f53b157b69da319e2f49ab4d', 1480886040.0)]
+
+        for x in range(len(expected)):
```
Comment (sduenas) [inline tests/test_rss.py]: You should check some of the elements of the parsed data (such as the title), not just the metadata fields.

File: tests/test_rss.py [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+        self.tmp_path = tempfile.mkdtemp(prefix='perceval_')
+
+    def tearDown(self):
+        shutil.rmtree(self.tmp_path)
+
+    @httpretty.activate
+    def test_fetch_from_cache(self):
+        """Test whether the cache works"""
+
+        bodies_entries = read_file('data/rss_entries.xml')
```
Comment (sduenas) [inline tests/test_rss.py]: This does the same of  `__configure_http_server()` method from `TestRSSBackend`.

File: tests/test_rss.py [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+                               ])
+
+        # First, we fetch the entries from the server, storing them
+        # in a cache
+        cache = Cache(self.tmp_path)
+        rss = RSS(RSS_FEED_URL, cache=cache)
+
+        entries = [item for item in rss.fetch()]
+
+        # Now, we get the entries from the cache.
```
Comment (sduenas) [inline tests/test_rss.py]: You should check the number of requests here, to know whether new ones were sent/received or not.

File: perceval/backends/core/rss.py [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+        logger.info("Looking for rss entries at feed '%s'", self.url)
+
+        self._purge_cache_queue()
+        nentries = 0  # number of entries
+
+        raw_entries = self.client.get_entries()
+        self._push_cache_queue(raw_entries)
+        self._flush_cache_queue()
+        entries = feedparser.parse(raw_entries)['entries']
```
Comment (sduenas) [inline perceval/backends/core/rss.py]: This also will need unit tests

File: perceval/backends/core/rss.py [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+        """
+
+        logger.info("Looking for rss entries at feed '%s'", self.url)
+
+        self._purge_cache_queue()
+        nentries = 0  # number of entries
+
+        raw_entries = self.client.get_entries()
+        self._push_cache_queue(raw_entries)
+        self._flush_cache_queue()
```
Comment (acs) [inline perceval/backends/core/rss.py]: Done!

File: perceval/backends/core/rss.py (hunk 10 above)
Comment (acs) [inline perceval/backends/core/rss.py]: Moved to a static method the parsing.

About the unit test, we are testing in this case the external library feedparser, right? Added!

File: perceval/backends/core/rss.py [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend does not supports entries resuming
+        """
+        return False
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a Build item."""
```
Comment (acs) [inline perceval/backends/core/rss.py]: Done!

File: perceval/backends/core/rss.py [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+        return str(item['link'])
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a RSS item.
+
+        The timestamp is extracted from 'timestamp' field.
```
Comment (acs) [inline perceval/backends/core/rss.py]: Done!

File: perceval/backends/core/rss.py (hunk 1 above)
Comment (acs) [inline perceval/backends/core/rss.py]: Yes, I had the same doubt. "entry" is the right category, I think. Changed!

File: tests/test_rss.py (hunk 4 above)
Comment (acs) [inline tests/test_rss.py]: Arrggg, fixed!

File: tests/test_rss.py (hunk 5 above)
Comment (acs) [inline tests/test_rss.py]: Yes! Fixed!

File: tests/test_rss.py (hunk 6 above)
Comment (acs) [inline tests/test_rss.py]: Yes, it was used only inside this class this is why I left it there but I have moved it outside the class.

File: tests/test_rss.py (hunk 7 above)
Comment (acs) [inline tests/test_rss.py]: Done!

File: tests/test_rss.py (hunk 8 above)
Comment (acs) [inline tests/test_rss.py]: Yes, reused this method! Fixed!

File: tests/test_rss.py (hunk 9 above)
Comment (acs) [inline tests/test_rss.py]: Done!

**Pull request closed without merging**